src/api/lib/backend/api/published.rb

```diff
@@ -17,8 +17,7 @@ def self.download_url_for_repository(project_name, repository_name, view = :publ
       def self.build_id(project_name, repository_name)
         Rails.cache.fetch("build_id-#{project_name}-#{repository_name}") do
           response = download_url_for_repository(project_name, repository_name, :status)
-          result = Xmlhash.parse(response).with_indifferent_access
-          result[:buildid]
+          (Xmlhash.parse(response) || {})['buildid']
```

Comment:
why do we need now `{}`? :confused:

Reply:
Xmlhash can return nil which would crash then.

Reply:
e.g. on 404s

Reply:
then this was already failing.... :confused:

Reply:
aehm, just checked - 404s throw an exception. So parse basically can't return nil - unless the backend is broken, which I don't think we want to avoid optimize for

Reply:
Dropped